fawltydeps/packages.py

```diff
@@ -180,44 +178,93 @@ def packages(self) -> Dict[str, Package]:
+
+        Provide a path to the temporary venv into the caller's context in which
+        the given requirements have been `pip install`ed. Automatically remove
+        the venv at the end of the context.
         """
-        return self.packages.get(Package.normalize_name(package_name))
+        with tempfile.TemporaryDirectory() as tmpdir:
+            venv_dir = Path(tmpdir)
+            venv.create(venv_dir, clear=True, with_pip=True)
+            subprocess.run(
+                [f"{venv_dir}/bin/pip", "install", "--no-deps"] + requirements,
+                check=True,  # fail if any of the commands fail
+            )
+            (venv_dir / ".installed").touch()
+            yield venv_dir
 
+    def lookup_packages(self, package_names: Set[str]) -> Dict[str, Package]:
+        """Convert package names into Package objects via temporary pip install.
+
+        Use the temp_installed_requirements() above to `pip install` the given
+        package names into a temporary venv, and then use LocalPackageResolver
+        on this venv to provide the Package objects that correspond to the
+        package names.
+        """
+        logger.info("Installing dependencies into a new temporary Python environment.")
+        with self.temp_installed_requirements(sorted(package_names)) as venv_dir:
+            local_resolver = LocalPackageResolver(venv_dir)
+            return local_resolver.lookup_packages(package_names)
 
-@contextmanager
-def temp_installed_requirements(
-    requirements: List[str],
-) -> Iterator[LocalPackageLookup]:
-    """Install packages into a temporary venv and provide lookup.
 
-    Provide a `LocalPackageLookup` object into the caller's context which
-    provide dependency-to-imports mapping for the given requirements.
-    The requirements are downloaded and installed into a temporary virtualenv,
-    which is automatically deleted at the end of this context.
+class IdentityMapping(BasePackageResolver):
+    """An imperfect package resolver that assumes package name == import name.
+
+    This will resolve _any_ package name into a corresponding identical import
+    name (modulo normalization, see Package.normalize_name() for details).
     """
-    with tempfile.TemporaryDirectory() as tmpdir:
-        venv_dir = Path(tmpdir)
-        venv.create(venv_dir, clear=True, with_pip=True)
-        subprocess.run(
-            [f"{venv_dir}/bin/pip", "install", "--no-deps"] + requirements,
-            check=True,  # fail if any of the commands fail
+
+    @staticmethod
+    def lookup_package(package_name: str) -> Package:
+        """Convert a package name into a Package with the same import name."""
+        ret = Package(package_name)
+        import_name = Package.normalize_name(package_name)
+        ret.add_import_names(import_name, mapping=DependenciesMapping.IDENTITY)
+        logger.info(
+            f"Could not find {package_name!r} in the current environment. "
```

Comment:
I think this should be reworded. Below is a lazy suggestion:
```suggestion
            f"{package_name!r} was not resolved. "
```

Reply:
A little more verbose:

```suggestion
            f"Failed to resolve {package_name!r} in the given Python environments. "
```

Reply:
My problem with the above:
- When a user mapping is provided, it's no longer just an issue of failing to resolve in a Python env.
- When identity mapping is the only mapping, we simply cannot say that we "failed" to resolve.

Again, this is for later, so we can open an issue to remember to review this log message when we're done with the mapping milestone.